### PR TITLE
add recordio to decode scheduler event

### PR DESCRIPTION
**Pull request by weiwei04**

### CODE REVIEW

**Comment by upccup** [comment]
👍

**Comment by pwzgorilla** [comment]
@weiwei04 这样做的好处是什么呢?只是增加了`reader buffer`吗?

**Comment by cmingxu** [comment]
@pwzgorilla I am not quite familiar with the mesos subscribed json response body  for now, But I think if mesos-go handle it like this, it might because each chunk of json body is not complete under extreme network conditions, we need stick them together and parse the event struct one by one.

So this PR definitely looks good to me, Thanks @weiwei04

**Comment by weiwei04** [comment]
@pwzgorilla @cmingxu 

根据[mesos scheduler http api](http://mesos.apache.org/documentation/latest/scheduler-http-api/)的说明，subscribe成功之后，后续读取的数据格式为 [uint64 payload lentght][\n][payload]，所以[mesos-go](https://github.com/mesos/mesos-go/tree/next/recordio)中实现了个reader来读取response。

swan中当前的实现虽然也可以work，但是会有很多decode失败的情况，猜测，也是因为这个原因swan的实现中只有io.EOF的错误才return。

**Comment by xiaods** [comment]
LGTM

File: scheduler/scheduler.go [hunk 1]
```diff
@@ -127,14 +126,15 @@ func (s *Scheduler) handleEvents(resp *http.Response) {
...
+
 	for {
 		event := new(sched.Event)
 		if err := dec.Decode(event); err != nil {
-			if err == io.EOF {
-				return
-			}
-			continue
+			logrus.Errorf("Deocde event failed: %s", err)
+			return
```
Comment (pwzgorilla) [inline scheduler/scheduler.go]: 这里是不是应该不是return呢？

Comment (pwzgorilla) [inline scheduler/scheduler.go]: @weiwei04 嗯，那这样的话，return其实也是对的。后续再做个错误处理的机制。

**Comment by pwzgorilla** [comment]
@weiwei04 thank you for so wonderful PR, thanks!

**Pull request merged**